packages/frontend/core/src/components/blocksuite/block-suite-mode-switch/index.tsx

```diff
@@ -1,128 +1,143 @@
+  const trash = useLiveData(doc?.trash$);
+  const currentMode = useLiveData(doc?.mode$);
 
-  const onSwitchToPageMode = useCallback(() => {
-    track.$.header.actions.switchPageMode({
-      mode: 'page',
-    });
-    if (currentMode === 'page' || isPublic) {
-      return;
-    }
-    doc.setMode('page');
+  const togglePage = useCallback(() => {
+    if (currentMode === 'page' || isPublic || trash) return;
+    doc?.setMode('page');
     toast(t['com.affine.toastMessage.pageMode']());
-  }, [currentMode, isPublic, doc, t]);
+    track.$.header.actions.switchPageMode({ mode: 'page' });
+  }, [currentMode, doc, isPublic, t, trash]);
 
-  const onSwitchToEdgelessMode = useCallback(() => {
-    track.$.header.actions.switchPageMode({
-      mode: 'edgeless',
-    });
-    if (currentMode === 'edgeless' || isPublic) {
-      return;
-    }
-    doc.setMode('edgeless');
+  const toggleEdgeless = useCallback(() => {
+    if (currentMode === 'edgeless' || isPublic || trash) return;
+    doc?.setMode('edgeless');
     toast(t['com.affine.toastMessage.edgelessMode']());
-  }, [currentMode, isPublic, doc, t]);
+    track.$.header.actions.switchPageMode({ mode: 'edgeless' });
+  }, [currentMode, doc, isPublic, t, trash]);
+
+  const onModeChange = useCallback(
+    (mode: DocMode) => {
+      mode === 'page' ? togglePage() : toggleEdgeless();
+    },
+    [toggleEdgeless, togglePage]
+  );
 
   const shouldHide = useCallback(
     (mode: DocMode) =>
       (trash && currentMode !== mode) || (isPublic && publicMode !== mode),
     [currentMode, isPublic, publicMode, trash]
   );
 
-  const shouldActive = useCallback(
-    (mode: DocMode) => (isPublic ? false : currentMode === mode),
-    [currentMode, isPublic]
-  );
+  useEffect(() => {
+    if (trash || isPublic || currentMode === undefined) return;
+    return registerAffineCommand({
+      id: 'affine:doc-mode-switch',
+      category: 'editor:page',
+      label:
+        currentMode === 'page'
+          ? t['com.affine.cmdk.switch-to-edgeless']()
+          : t['com.affine.cmdk.switch-to-page'](),
+      icon: currentMode === 'page' ? <EdgelessIcon /> : <PageIcon />,
+      keyBinding: {
+        binding: 'Alt+KeyS',
+        capture: true,
+      },
+      run: () => onModeChange(currentMode === 'edgeless' ? 'page' : 'edgeless'),
+    });
+  }, [currentMode, isPublic, onModeChange, t, trash]);
 
   return (
     <Tooltip
       content={t['Switch']()}
       shortcut={['$alt', 'S']}
       side="bottom"
-      options={{
-        hidden: isPublic || trash,
-      }}
+      options={{ hidden: isPublic || trash }}
     >
-      <StyledEditorModeSwitch
-        style={style}
-        switchLeft={currentMode === 'page'}
-        showAlone={trash || isPublic}
-      >
-        <PageSwitchItem
-          data-testid="switch-page-mode-button"
-          active={shouldActive('page')}
-          hide={shouldHide('page')}
-          trash={trash}
-          onClick={onSwitchToPageMode}
+      <div>
```

Comment:
useless div?

Reply:
We use `asChild` for `Tooltip.Target`, so the children of `Tooltip` should be able to bind `ref`.